### DISCUSSION TITLE
Call submit self-service recovery with token

### DISCUSSION THREAD

**Discussion by Asilent** [ory/kratos#2756] [Q&A] [answered]
I use kratos client to call `SubmitSelfServiceRecoveryFlow` with token sent by recovery-emaiI. I expect a response with a session token, but what I got was a rediret to login instead of setting.
I checked the Docker logs and I found `ExecutePostVerificationHook completed successfully` in it.
However, below this log is an error about /session/whoami, and these errors affect the response of the client API.
Like the image below:
<img width="1493" alt="docker log" src="https://user-images.githubusercontent.com/44249329/192098884-4e0c9b10-20c3-4e22-a84c-6b26886fb263.png">
This representation is different from the verification process.

**Comment by vinckr** [comment, marked as answer]
Hello @Asilent 
Apologies for the late response!

It seems like you're trying to use the Ory Kratos client to submit a recovery flow. However, you're encountering an issue where you're being redirected to login instead of receiving a session token.
The Ory Session Token can be checked at the http://127.0.0.1/sessions/whoami endpoint. If there's an error related to this endpoint in your Docker logs, it might be affecting the response of the client API.

The recovery flow involves submitting the user's email via a POST request to the /self-service/recovery endpoint. For browser applications, you must send all cookies and the CSRF token in the request body. The CSRF token value is a hidden input field called csrf_token.

```
curl -X POST -H 'Content-Type: application/json' \  
 -H 'Accept: application/json' \  
 -d '{"method":"code","email":"[[email protected]](https://www.ory.sh/cdn-cgi/l/email-protection)","csrf_token":"your-csrf-token"}' \  
 -b cookies.txt \  
 "https://{project.slug}.projects.oryapis.com/self-service/recovery?flow=<your-flow-id>"
```

If you're still encountering issues, you might want to check the [Ory Kratos User Login](https://www.ory.sh/docs/kratos/self-service/flows/user-login) and [User Registration Documentation](https://www.ory.sh/docs/kratos/self-service/flows/user-registration) for more information.

However, without more specific information about the error you're seeing, it's difficult to provide a more precise answer. 

PS: due to the age of the discussion I am marking it as answered now but feel free to revive the thread at any time!
